src/Ucommerce.Avenue.Umbraco/Api/AvenueClothingApiSearchController.cs

```diff
@@ -6,35 +6,32 @@
 using UCommerce.Infrastructure;
 using UCommerce.RazorStore.Api.Model;
 using UCommerce.Search;
+using UCommerce.Search.Models;
 using UCommerce.SystemHttp.Models;
 
 namespace UCommerce.RazorStore.Api
 {
     [RoutePrefix("ucommerceapi")]
     public class AvenueClothingApiSearchController : ApiController
     {
-        public ISlugService UrlService => ObjectFactory.Instance.Resolve<ISlugService>();
-        public ICatalogLibrary CatalogLibrary => ObjectFactory.Instance.Resolve<ICatalogLibrary>();
+        public IUrlService UrlService => ObjectFactory.Instance.Resolve<IUrlService>();
         public ICatalogContext CatalogContext => ObjectFactory.Instance.Resolve<ICatalogContext>();
+        public IIndex<Product> ProductIndex => ObjectFactory.Instance.Resolve<IIndex<Product>>();
 
         [Route("razorstore/search/")]
         [HttpPost]
         public IHttpActionResult Search([FromBody] SearchRequest request)
         {
-            // TODO: 
-            
-            var products = UCommerce.EntitiesV2.Product.Find(p =>
-                p.VariantSku == null
-                && p.DisplayOnSite
-                &&
-                (
-                    p.Sku.Contains(request.Keyword)
-                    || p.Name.Contains(request.Keyword)
-                    || p.ProductDescriptions.Any(d =>
-                        d.DisplayName.Contains(request.Keyword) || d.ShortDescription.Contains(request.Keyword) ||
-                        d.LongDescription.Contains(request.Keyword))
-                )
-            );
+            var products = ProductIndex.Find()
+                .Where(p => p.VariantSku == null
+                            && (
+                                p.Sku.Contains(request.Keyword)
+                                || p.Name.Contains(request.Keyword)
+                                || p.DisplayName.Contains(request.Keyword)
+                                || p.LongDescription.Contains(request.Keyword)
+                                || p.ShortDescription.Contains(request.Keyword)
+                            ))
+                .ToList();
```

Comment:
It's great that the query translates so 1:1 from Linq to Bolt! 😉 
However, shouldn't we rather use `p.DisplayName == Match.Fulltext(request.Keyword)` in order to use the fact that we full text index some of the fields?

Reply:
Yep, my bad.